src/linear_algebra/matrix/hermitian.lean

```diff
@@ -139,34 +109,91 @@ lemma is_hermitian_from_blocks_iff
+(such as on the reals). -/
+@[simp] lemma is_hermitian_diagonal [has_trivial_star α] [decidable_eq n] (v : n → α) :
+  (diagonal v).is_hermitian :=
+-- TODO: add a `pi.has_trivial_star` instance and remove the `funext`
+(diagonal_conj_transpose v).trans $ congr_arg _ $ funext $ λ i, star_trivial _
 
-@[simp] lemma is_hermitian_one [decidable_eq n] :
-  (1 : matrix n n α).is_hermitian :=
-conj_transpose_one
+@[simp] lemma is_hermitian_zero :
+  (0 : matrix n n α).is_hermitian :=
+is_self_adjoint_zero _
 
-end semiring
+@[simp] lemma is_hermitian.add {A B : matrix n n α} (hA : A.is_hermitian) (hB : B.is_hermitian) :
+  (A + B).is_hermitian :=
+is_self_adjoint.add hA hB
 
-section ring
+end add_monoid
 
-variables [ring α] [star_ring α] [ring β] [star_ring β]
+section add_comm_monoid
+variables [add_comm_monoid α] [star_add_monoid α]
+
+lemma is_hermitian_add_transpose_self (A : matrix n n α) :
+  (A + Aᴴ).is_hermitian :=
+is_self_adjoint_add_star_self A
+
+lemma is_hermitian_transpose_add_self (A : matrix n n α) :
+  (Aᴴ + A).is_hermitian :=
+is_self_adjoint_star_add_self A
+
+end add_comm_monoid
+
+section add_group
+variables [add_group α] [star_add_monoid α]
 
 @[simp] lemma is_hermitian.neg {A : matrix n n α} (h : A.is_hermitian) :
   (-A).is_hermitian :=
-(conj_transpose_neg _).trans (congr_arg _ h)
+is_self_adjoint.neg h
 
 @[simp] lemma is_hermitian.sub {A B : matrix n n α} (hA : A.is_hermitian) (hB : B.is_hermitian) :
   (A - B).is_hermitian :=
-(conj_transpose_sub _ _).trans (hA.symm ▸ hB.symm ▸ rfl)
+is_self_adjoint.sub hA hB
 
-end ring
+end add_group
 
-section comm_ring
+section non_unital_semiring
+variables [non_unital_semiring α] [star_ring α] [non_unital_semiring β] [star_ring β]
+
+lemma is_hermitian_mul_conj_transpose_self [fintype n] (A : matrix n n α) :
+  (A ⬝ Aᴴ).is_hermitian :=
+is_self_adjoint.mul_star_self A
+
+lemma is_hermitian_transpose_mul_self [fintype n] (A : matrix n n α) :
+  (Aᴴ ⬝ A).is_hermitian :=
+is_self_adjoint.star_mul_self A
 
+/-- Note this is more general than `is_self_adjoint.conjugate'` as `B` can be rectangular. -/
+lemma is_hermitian_conj_transpose_mul_mul [fintype m] {A : matrix m m α} (B : matrix m n α)
+  (hA : A.is_hermitian) : (Bᴴ ⬝ A ⬝ B).is_hermitian :=
+by simp only [is_hermitian, conj_transpose_mul, conj_transpose_conj_transpose, hA.eq,
+  matrix.mul_assoc]
+
+/-- Note this is more general than `is_self_adjoint.conjugate` as `B` can be rectangular. -/
+lemma is_hermitian_mul_mul_conj_transpose [fintype m] {A : matrix m m α} (B : matrix n m α)
+  (hA : A.is_hermitian) : (B ⬝ A ⬝ Bᴴ).is_hermitian :=
+by simp only [is_hermitian, conj_transpose_mul, conj_transpose_conj_transpose, hA.eq,
+  matrix.mul_assoc]
+
+end non_unital_semiring
+
+section semiring
+
+variables [semiring α] [star_ring α] [semiring β] [star_ring β]
+
+/-- Note this is more generally than `is_self_adjoint_one` as it does not require `fintype n`. -/
```

Comment:
I don't understand this comment at all. Where does `is_self_adjoint_one` require `fintype n`?

Reply:
`is_self_adjoint_one` requires `fintype n` because it requires `star_ring (matrix n n R)` which requires `ring (matrix n n R)`

Reply:
Aha, the point is we don't have a (globallly defined) matrix multiplication; got it. Maybe you can say that instead (and fix the typo):
```suggestion
/-- Note this is more general for matrices than `is_self_adjoint_one` as it does not
require `fintype n`, which is necessary for `monoid (matrix n n R)`. -/
```

Reply:
LGTM, thanks!